rclcpp/src/rclcpp/node.cpp

```diff
@@ -58,6 +59,12 @@ Node::Node(
       node_topics_.get(),
       use_intra_process_comms
     )),
+  node_clock_(new rclcpp::node_interfaces::NodeClock(
+      node_base_,
+      node_topics_,
+      node_graph_,
+      node_services_
+    )),
```

Comment:
This line currently segfaults with `rmw_opensplice_cpp` which fails all kinds of communication tests (see http://ci.ros2.org/job/ci_linux/3696/testReport/). This can be easily reproduced by running the `demo_nodes_cpp` `listener`.

The segfault:

```
terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
  what():  could not create subscription: type support handle implementation 'rosidl_typesupport_cpp' (0x7f26431938d0) does not match valid type supports ('rosidl_typesupport_opensplice_cpp' (0x7f2641c5e5a0), 'rosidl_typesupport_opensplice_c' (0x7f2642066740)), at .../src/ros2/rmw_opensplice/rmw_opensplice_cpp/src/rmw_subscription.cpp:63, at .../src/ros2/rcl/rcl/src/rcl/subscription.c:149
```

The stacktrace is rather lengthy:

```
#0  0x00007ffff6790428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff679202a in __GI_abort () at abort.c:89
#2  0x00007ffff6dca84d in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff6dc86b6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff6dc8701 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff6dc8919 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff784ac72 in rclcpp::exceptions::throw_from_rcl_error (ret=1, prefix="could not create subscription", error_state=0x662130, reset_error=0x40b020 <rcutils_reset_error@plt>)
    at .../ros2/rclcpp/rclcpp/src/rclcpp/exceptions.cpp:72
#7  0x00007ffff78f6618 in rclcpp::subscription::SubscriptionBase::SubscriptionBase (this=0x658890, node_handle=std::shared_ptr (expired, weak 0) 0x7fffffff5510, type_support_handle=..., topic_name="/clock", 
    subscription_options=...) at .../ros2/rclcpp/rclcpp/src/rclcpp/subscription.cpp:54
#8  0x00007ffff790c08c in rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >::Subscription (this=0x658890, 
    node_handle=std::shared_ptr (expired, weak 0) 0x7fffffff55a0, topic_name="/clock", subscription_options=..., callback=..., memory_strategy=std::shared_ptr (expired, weak 0) 0x7fffffff55b0)
    at .../ros2/rclcpp/rclcpp/include/rclcpp/subscription.hpp:166
#9  0x00007ffff790bc2e in __gnu_cxx::new_allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >::construct<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >, std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >*, std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) (
    this=0x7fffffff5867, __p=0x658890) at /usr/include/c++/5/ext/new_allocator.h:120
#10 0x00007ffff790b293 in std::allocator_traits<std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > >::construct<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >, std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >&, rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >*, std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) (__a=..., __p=0x658890) at /usr/include/c++/5/bits/alloc_traits.h:530
#11 0x00007ffff790a537 in std::_Sp_counted_ptr_inplace<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >, std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >, std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) (this=0x658880, __a=...) at /usr/include/c++/5/bits/shared_ptr_base.h:522
#12 0x00007ffff790908d in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >, std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >, std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::_Sp_make_shared_tag, rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >*, std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&, std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) (
    this=0x7fffffff5bf8, __a=...) at /usr/include/c++/5/bits/shared_ptr_base.h:617
#13 0x00007ffff7907ab4 in std::__shared_ptr<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >, std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::_Sp_make_shared_tag, std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&, std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) (this=0x7fffffff5bf0, __tag=..., __a=...) at /usr/include/c++/5/bits/shared_ptr_base.h:1097
#14 0x00007ffff7905b2e in std::shared_ptr<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >::shared_ptr<std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >, std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::_Sp_make_shared_tag, std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&, std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) (this=0x7fffffff5bf0, __tag=..., __a=...) at /usr/include/c++/5/bits/shared_ptr.h:319
#15 0x00007ffff7903508 in std::allocate_shared<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >, std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >, std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::allocator<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&, std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_---Type <return> to continue, or q <return> to quit---
interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) (__a=...) at /usr/include/c++/5/bits/shared_ptr.h:620
#16 0x00007ffff7900d3a in std::make_shared<rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >, std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) ()
    at /usr/include/c++/5/bits/shared_ptr.h:636
#17 0x00007ffff78fdd7e in rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >::make_shared<std::shared_ptr<rcl_node_t>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&>(std::shared_ptr<rcl_node_t>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&, rclcpp::any_subscription_callback::AnySubscriptionCallback<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > const&, std::shared_ptr<rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > > const&) ()
    at .../ros2/rclcpp/rclcpp/include/rclcpp/subscription.hpp:138
#18 0x00007ffff78fbb0a in rclcpp::SubscriptionFactory rclcpp::create_subscription_factory<builtin_interfaces::msg::Time_<std::allocator<void> >, std::_Bind<std::_Mem_fn<void (rclcpp::TimeSource::*)(std::shared_ptr<builtin_interfaces::msg::Time_<std::allocator<void> > >)> (rclcpp::TimeSource*, std::_Placeholder<1>)>, std::allocator<void>, rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >(std::_Bind<std::_Mem_fn<void (rclcpp::TimeSource::*)(std::shared_ptr<builtin_interfaces::msg::Time_<std::allocator<void> > >)> (rclcpp::TimeSource*, std::_Placeholder<1>)>&&, rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >::SharedPtr, std::shared_ptr<std::allocator<void> >)::{lambda(rclcpp::node_interfaces::NodeBaseInterface*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&)#1}::operator()(rclcpp::node_interfaces::NodeBaseInterface*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&) const (__closure=0x660130, node_base=0x63ca00, topic_name="/clock", subscription_options=...)
    at .../ros2/rclcpp/rclcpp/include/rclcpp/subscription_factory.hpp:104
#19 0x00007ffff7903663 in std::_Function_handler<std::shared_ptr<rclcpp::subscription::SubscriptionBase> (rclcpp::node_interfaces::NodeBaseInterface*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&), rclcpp::SubscriptionFactory rclcpp::create_subscription_factory<builtin_interfaces::msg::Time_<std::allocator<void> >, std::_Bind<std::_Mem_fn<void (rclcpp::TimeSource::*)(std::shared_ptr<builtin_interfaces::msg::Time_<std::allocator<void> > >)> (rclcpp::TimeSource*, std::_Placeholder<1>)>, std::allocator<void>, rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >(std::_Bind<std::_Mem_fn<void (rclcpp::TimeSource::*)(std::shared_ptr<builtin_interfaces::msg::Time_<std::allocator<void> > >)> (rclcpp::TimeSource*, std::_Placeholder<1>)>&&, rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >::SharedPtr, std::shared_ptr<std::allocator<void> >)::{lambda(rclcpp::node_interfaces::NodeBaseInterface*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&)#1}>::_M_invoke(std::_Any_data const&, rclcpp::node_interfaces::NodeBaseInterface*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&) (__functor=..., 
    __args#0=<unknown type in .../install_isolated/rclcpp/lib/librclcpp.so, CU 0x355988, DIE 0x3a3cea>, __args#1="/clock", __args#2=...) at /usr/include/c++/5/functional:1857
#20 0x00007ffff789d7b1 in std::function<std::shared_ptr<rclcpp::subscription::SubscriptionBase> (rclcpp::node_interfaces::NodeBaseInterface*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&)>::operator()(rclcpp::node_interfaces::NodeBaseInterface*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcl_subscription_options_t&) const (this=0x7fffffff5e90, __args#0=0x63ca00, __args#1="/clock", __args#2=...) at /usr/include/c++/5/functional:2267
#21 0x00007ffff789cc85 in rclcpp::node_interfaces::NodeTopics::create_subscription (this=0x656460, topic_name="/clock", subscription_factory=..., subscription_options=..., use_intra_process=false)
    at .../ros2/rclcpp/rclcpp/src/rclcpp/node_interfaces/node_topics.cpp:90
#22 0x00007ffff78fa541 in rclcpp::create_subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::_Bind<std::_Mem_fn<void (rclcpp::TimeSource::*)(std::shared_ptr<builtin_interfaces::msg::Time_<std::allocator<void> > >)> (rclcpp::TimeSource*, std::_Placeholder<1>)>, std::allocator<void>, rclcpp::subscription::Subscription<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> > >(rclcpp::node_interfaces::NodeTopicsInterface*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::_Bind<std::_Mem_fn<void (rclcpp::TimeSource::*)(std::shared_ptr<builtin_interfaces::msg::Time_<std::allocator<void> > >)> (rclcpp::TimeSource*, std::_Placeholder<1>)>&&, rmw_qos_profile_t const&, std::shared_ptr<rclcpp::callback_group::CallbackGroup>, bool, bool, rclcpp::message_memory_strategy::MessageMemoryStrategy<builtin_interfaces::msg::Time_<std::allocator<void> >, std::allocator<void> >::SharedPtr, std::shared_ptr<std::allocator<void> >) (node_topics=0x656460, topic_name="/clock", 
    callback=<unknown type in .../install_isolated/rclcpp/lib/librclcpp.so, CU 0x355988, DIE 0x397949>, qos_profile=..., group=std::shared_ptr (empty) 0x0, ignore_local_publications=false, 
    use_intra_process_comms=false, msg_mem_strat=std::shared_ptr (count 4, weak 0) 0x661370, allocator=std::shared_ptr (count 3, weak 0) 0x655490)
    at .../ros2/rclcpp/rclcpp/include/rclcpp/create_subscription.hpp:55
#23 0x00007ffff78f7edf in rclcpp::TimeSource::attachNode (this=0x6586f8, node_base_interface=std::shared_ptr (count 5, weak 0) 0x63ca00, node_topics_interface=std::shared_ptr (count 5, weak 0) 0x656460, 
    node_graph_interface=std::shared_ptr (count 5, weak 0) 0x656cf0, node_services_interface=std::shared_ptr (count 5, weak 0) 0x655550)
    at .../ros2/rclcpp/rclcpp/src/rclcpp/time_source.cpp:77
#24 0x00007ffff787c57e in rclcpp::node_interfaces::NodeClock::NodeClock (this=0x6586a0, node_base=std::shared_ptr (count 6539776, weak -1) 0x7fffffff6140, 
    node_topics=std::shared_ptr (count 6585088, weak -1) 0x7fffffff6150, node_graph=std::shared_ptr (count 6640944, weak -1) 0x7fffffff6160, node_services=std::shared_ptr (count 6563888, weak -1) 0x7fffffff6170)
    at .../ros2/rclcpp/rclcpp/src/rclcpp/node_interfaces/node_clock.cpp:37
#25 0x00007ffff78770ed in rclcpp::node::Node::Node (this=0x63b690, node_name="listener", namespace_="", context=std::shared_ptr (expired, weak 0) 0x7fffffff61f0, use_intra_process_comms=false)
    at .../ros2/rclcpp/rclcpp/src/rclcpp/node.cpp:67
#26 0x00007ffff7876dc5 in rclcpp::node::Node::Node (this=0x63b690, node_name="listener", namespace_="", use_intra_process_comms=false) at .../ros2/rclcpp/rclcpp/src/rclcpp/node.cpp:45
#27 0x00000000004145d0 in __gnu_cxx::new_allocator<rclcpp::node::Node>::construct<rclcpp::node::Node, char const (&) [9]> (this=0x7fffffff6397, __p=0x63b690) at /usr/include/c++/5/ext/new_allocator.h:120
#28 0x0000000000413aa0 in std::allocator_traits<std::allocator<rclcpp::node::Node> >::construct<rclcpp::node::Node, char const (&) [9]> (__a=..., __p=0x63b690) at /usr/include/c++/5/bits/alloc_traits.h:530
#29 0x0000000000412858 in std::_Sp_counted_ptr_inplace<rclcpp::node::Node, std::allocator<rclcpp::node::Node>, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<char const (&) [9]> (this=0x63b680, __a=...)
    at /usr/include/c++/5/bits/shared_ptr_base.h:522
#30 0x0000000000411129 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<rclcpp::node::Node, std::allocator<rclcpp::node::Node>, char const (&) [9]> (this=0x7fffffff6588, __a=...)
    at /usr/include/c++/5/bits/shared_ptr_base.h:617
#31 0x000000000040fae0 in std::__shared_ptr<rclcpp::node::Node, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<rclcpp::node::Node>, char const (&) [9]> (this=0x7fffffff6580, __tag=..., __a=...)
    at /usr/include/c++/5/bits/shared_ptr_base.h:1097
#32 0x000000000040e1d8 in std::shared_ptr<rclcpp::node::Node>::shared_ptr<std::allocator<rclcpp::node::Node>, char const (&) [9]> (this=0x7fffffff6580, __tag=..., __a=...) at /usr/include/c++/5/bits/shared_ptr.h:319
#33 0x000000000040ceb0 in std::allocate_shared<rclcpp::node::Node, std::allocator<rclcpp::node::Node>, char const (&) [9]> (__a=...) at /usr/include/c++/5/bits/shared_ptr.h:620
---Type <return> to continue, or q <return> to quit---
#34 0x000000000040c5cd in std::make_shared<rclcpp::node::Node, char const (&) [9]> () at /usr/include/c++/5/bits/shared_ptr.h:636
#35 0x000000000040c030 in rclcpp::node::Node::make_shared<char const (&) [9]> () at .../install_isolated/rclcpp/include/rclcpp/node.hpp:66
#36 0x000000000040b700 in main (argc=1, argv=0x7fffffff6718) at .../ros2/demos/demo_nodes_cpp/src/topics/listener.cpp:41
```

This needs to be fixed asap for the release.

Reply:
The somehow good news is that the list of failing tests looks very similar to the failures we had with opensplice on OSX. We may be able to release opensplice on all platforms once this is fixed

Reply:
I do not believe this is a segmentation fault, I think it's just an exception.

Reply:
Underlying issue ticketed at: https://github.com/ros2/rmw_opensplice/issues/216